lib/iris/common/metadata.py

```diff
@@ -200,6 +725,107 @@ class CoordMetadata(BaseMetadata):
 
     __slots__ = ()
 
+    @wraps(BaseMetadata.__eq__, assigned=("__doc__",), updated=())
+    @lenient_service
+    def __eq__(self, other):
+        return super().__eq__(other)
+
+    def _combine_lenient(self, other):
+        """
+        Perform lenient combination of metadata members for coordinates.
+
+        Args:
+
+        * other (CoordMetadata):
+            The other coordinate metadata participating in the lenient
+            combination.
+
+        Returns:
+            A list of combined metadata member values.
+
+        """
+        # Perform "strict" combination for "coord_system" and "climatological".
+        def func(field):
+            left = getattr(self, field)
+            right = getattr(other, field)
+            return left if left == right else None
+
+        # Note that, we use "_members" not "_fields".
+        values = [func(field) for field in self._members]
+        # Perform lenient combination of the other parent members.
+        result = super()._combine_lenient(other)
+        result.extend(values)
+
+        return result
+
+    def _compare_lenient(self, other):
+        """
+        Perform lenient equality of metadata members for coordinates.
+
+        Args:
+
+        * other (CoordMetadata):
+            The other coordinate metadata participating in the lenient
+            comparison.
+
+        Returns:
+            Boolean.
+
+        """
+        # Perform "strict" comparison for "coord_system" and "climatological".
+        result = (
+            self.coord_system == other.coord_system
+            and self.climatological == other.climatological
+        )
```

Comment:
Now this is getting interesting, and to the heart of the question "what should be lenient?"

Personally, I think that `climatological` should be compared strictly in lenient mode - I guess simply because it's boolean, and mixing in leniency doesn't really help matters, so keep it simple.

For `coord_system` I flipped on this a few times. But at the end of the day I think that it's a pretty important core piece of metadata that makes a massive difference to interpreting the associated spatial coordinates. It think that its so important that users should always specify it and not rely on defaults or assumptions. We've certainly not made any statements regarding the default `coord_system` if it's not specified. So in that regard I erred on the side of comparing strictly in lenient mode - it seems like a non-controversial, safe stance to take.

Reply:
I went explicit when comparing, but happy to be more generic as you suggest. I've applied that pattern elsewhere, so why not here :+1: